packages/notebook/src/actions.tsx

```diff
@@ -2615,6 +2615,19 @@ namespace Private {
         });
         return;
       }
+      if (child.model.getMetadata('editable') == false) {
+        translator = translator || nullTranslator;
+        const trans = translator.load('jupyterlab');
+        // Do not permit changing cell type when the cell is readonly
+        void showDialog({
+          title: trans.__('Cell is not editable'),
+          body: trans.__(
+            'The cell is not editable, its type cannot be changed!'
+          ),
```

Comment:
Having looked more at the interface I wonder if it should be:

```suggestion
          title: trans.__('Cell is read-only'),
          body: trans.__(
            'The cell is read-only, its type cannot be changed!'
          ),
```

This is how we refer to non-editable cells in the property inspector:

![image](https://github.com/user-attachments/assets/0198aaad-f17f-4c4b-9a4f-4eceee566aaa)

and in the settings:

![image](https://github.com/user-attachments/assets/6fcc9f7c-367b-4e81-a2bd-3e113a7f495b)